sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs

```diff
@@ -795,6 +798,94 @@ static void GenerateNativeTypeNameAttribute(PInvokeGenerator generator, Stream?
             }
         }
 
+        static void GenerateNativeAnnotationAttribute(PInvokeGenerator generator, Stream? stream, bool leaveStreamOpen)
+        {
+            var config = generator.Config;
+
+            if (!generator._hasAnnotateAttr)
+            {
+                return;
+            }
```

Comment:
This is potentially problematic for scenarios where the developer generates helper types independently of any header that provides annotations.

This is done in various downstream tools where the tool is invoked once per header file in a library (rather than just once with all headers being processed by that single invocation).

I think it would be better to generate it by default and have a check for `--exclude NativeAnnotationAttribute` to support users who don't want it generated instead.

Reply:
Good idea. I implemented this in https://github.com/dotnet/ClangSharp/pull/552/commits/c62fb6231888351dad7e8a95001f1be58421cf17